mobile/lib/infrastructure/repositories/sync_api.repository.dart

```diff
@@ -70,24 +72,37 @@ class SyncApiRepository implements ISyncApiRepository {
       }
 
       await for (final chunk in response.stream.transform(utf8.decoder)) {
+        if (shouldAbort) {
+          _logger.warning("aborting stream");
+          break;
+        }
+
         previousChunk += chunk;
         final parts = previousChunk.toString().split('\n');
         previousChunk = parts.removeLast();
         lines.addAll(parts);
 
-        if (lines.length < _batchSize) {
+        if (lines.length < batchSize) {
           continue;
         }
 
-        yield _parseSyncResponse(lines);
+        await onData(_parseSyncResponse(lines), () => shouldAbort = true);
```

Comment:
probably just put this lamda into a private function and add a log in it saying "Abort requested, stopping sync stream" or similar.